packages/jest-mock/__typetests__/mock-functions.test.ts

```diff
@@ -55,6 +55,18 @@ expectType<Mock<() => string>>(
     .mockReturnValueOnce('value'),
 );
 
+expectType<Mock<() => undefined>>(
+  fn(() => undefined)
+    .mockImplementation()
+    .mockImplementationOnce()
+    .mockReturnValue()
+    .mockReturnValueOnce()
+    .mockResolvedValue()
+    .mockResolvedValueOnce()
+    .mockRejectedValue()
+    .mockRejectedValueOnce(),
+);
```

Comment:
Let’s extend this a bit (;

```suggestion
expectType<Mock<() => Promise<void>>>(
  fn(async () => {})
    .mockImplementation()
    .mockImplementationOnce()
    .mockResolvedValue()
    .mockResolvedValueOnce()
    .mockRejectedValue()
    .mockRejectedValueOnce()
    .mockReturnValue()
    .mockReturnValueOnce(),
);

expectType<Mock<() => void>>(
  fn(() => {})
    .mockImplementation()
    .mockImplementationOnce()
    .mockReturnValue()
    .mockReturnValueOnce(),
);
```